packages/@react-stately/layout/src/TableLayout.ts

```diff
@@ -10,50 +10,331 @@
  * governing permissions and limitations under the License.
  */
 
+import {
+  calculateColumnSizes,
+  getMaxWidth,
+  getMinWidth,
+  isStatic,
+  parseFractionalUnit
+} from './TableUtils';
+import {ColumnSize, TableCollection} from '@react-types/table';
 import {GridNode} from '@react-types/grid';
 import {Key} from 'react';
 import {LayoutInfo, Point, Rect, Size} from '@react-stately/virtualizer';
 import {LayoutNode, ListLayout, ListLayoutOptions} from './ListLayout';
-import {TableCollection} from '@react-types/table';
 
+type TableLayoutOptions<T> = ListLayoutOptions<T> & {
+  columnLayout: TableColumnLayout<T>,
+  initialCollection: TableCollection<T>
+}
+
+interface TableColumnLayoutOptions {
+  getDefaultWidth: (props) => ColumnSize,
+  getDefaultMinWidth: (props) => ColumnSize
+}
+
+export class TableColumnLayout<T> {
```

Comment:
I do wonder if this should be in `@react-stately/table`. So far the layout package has only been used by virtualizer.

Reply:
that makes sense, i'll see if it's easy to do